### DISCUSSION TITLE
WiFiNINA_Generic v1.8.14-5 Communication with WiFi module failed!

### DISCUSSION THREAD

**Discussion by Givy** [khoih-prog/WiFiNINA_Generic#20] [Ideas]
Hi,
Can "_SAMD51 ADAFRUIT_GRAND_CENTRAL_M4_" pin setting be added to the "_WiFiNINA_Pinout_Generic.h_" to work with the _ADAFRUIT_Airlift_?

Thank you.

**Comment by khoih-prog** [comment]
Thanks for your idea.

The obstacle is that I don't have that `ADAFRUIT_GRAND_CENTRAL_M4` board + `Airlift` to test. But it's easy for you to add some short and simple code to support the combinations.

Check 

https://github.com/khoih-prog/WiFiNINA_Generic/blob/a239f8e4269b3252873131e23bea31684f0e93bf/src/WiFiNINA_Pinout_Generic.h#L244-L259


Then you can add something similar, to the `WiFiNINA_Pinout_Generic.h` from line #259, such as

```
    #elif ( defined(ADAFRUIT_GRAND_CENTRAL_M4)  )  
      
      #warning You have to modify pin usage according to actual connection for SAMD51 DADAFRUIT_GRAND_CENTRAL_M4
      // To define pin out for WiFiNINA here


      //#define PINS_COUNT           (60u)
      //NINA     
      #define NINA_GPIO0      ESP32_GPIO0
      #define NINA_RESETN     ESP32_RESETN
      #define NINA_ACK        SPIWIFI_ACK


      #define SPIWIFI_SS      33           //  <=========== You have to modify here
      #define SPIWIFI_ACK     31           //  <=========== You have to modify here
      #define SPIWIFI_RESET   ESP32_RESETN           //  <=========== You have to modify here
      
    #else
```


Good Luck,
